PathPlanning/AStar/A_Star_searching_from_two_side.py

```diff
@@ -0,0 +1,348 @@
+'''
+A* algorithm
+Author: Weicent
+randomly generate obstacles, start and goal point
+searching path from start and end simultaneously
+'''
+
+import numpy as np
+import matplotlib.pyplot as plt
+import math
+
+show_animation = True
+
+
+class Node:
+    '''node with properties of g, h, coordinate and parent node'''
+
+    def __init__(self, G=0, H=0, coordinate=None, parent=None):
+        self.G = G
+        self.H = H
+        self.F = G + H
+        self.parent = parent
+        self.coordinate = coordinate
+
+    def setg(self, value):
```

Comment:
I think these setter functions are not needed for understanding algorithm. You can just access these fields.